Feature/cvtripleitem

LLdaniel opened the pull request:
Proposal for `cvtripleitem` feature request in #174: 
This PR can be used as a basis for discussion regarding the feature request.

djrmarques commented:
Thanks for opening this PR :) . I cannot merge it, but I hope it gets merged :smile:

LLdaniel commented:
Hi @jalopezg-git ,
thanks for your offer to review more PRs! I would be glad about a second opinion on that topic here. But also feel free to have a look at other PRs. Maybe we can also try to close the topics for the set milestone.
I opened it in February and at the moment I think why not implement it. Better having the option of a triple item than not even when people already asked for it. 

Because we delivered a few features in the last weeks I should probably catch up with the following points here:
- [ ] CHANGELOG
- [ ] proper documentation
- [ ] including your "contemporary" style in this PR
- [ ]  defining `\listitemsymbolspace` for the spacing like we have done for `cvlistitem` and `cvlistdoubleitem`

I will try to add that in the next few days.
What do you think?

LLdaniel commented on moderncvbodyi.sty:
```diff
@@ -60,6 +62,8 @@
   \setlength{\listitemcolumnwidth}{\maincolumnwidth-\listitemsymbolwidth}%
   \setlength{\doubleitemcolumnwidth}{\maincolumnwidth-\hintscolumnwidth-\separatorcolumnwidth-\separatorcolumnwidth}%
   \setlength{\doubleitemcolumnwidth}{0.5\doubleitemcolumnwidth}%
+  \setlength{\tripleitemcolumnwidth}{\maincolumnwidth-\hintscolumnwidth-\hintscolumnwidth-\separatorcolumnwidth-\separatorcolumnwidth-\separatorcolumnwidth}%
```
Hi @jalopezg-git ,
thanks - as always 😀 - for your review. I am happy to get some tasks done here. 
>> (I saw the answer below when reading the code - perhaps it's worth adding a comment here).

I tried to follow the logic of `\cvdoubleitem`. There is also a re-definition of `\doubleitemcolumnwidth` in the following line. 
Now as you pointed it out I also see that this could be confusing, because one would think that the first definition should be the final result of `\tripleitemcolumnwidth`. 
Unfortunately I am a bit stuck here in this view at the moment. But maybe there you could come into play with a fresh new sight:
Which kind of comment would have helped you here? I am glad to insert one, but right at the moment I am a bit lost what to write for clarification. 😅

>Other than that, cannot we just use `3\separatorcolumnwidth` instead of `-\separatorcolumnwidth-\separatorcolumnwidth-\separatorcolumnwidth`?

I agree, maybe I should have thought about using a factor earlier... 
Looks more like a simplified equation now. 🤣

jalopezg-git replied:
> I tried to follow the logic of `\cvdoubleitem`. There is also a re-definition of `\doubleitemcolumnwidth` in the following line. Now as you pointed it out I also see that this could be confusing, because one would think that the first definition should be the final result of `\tripleitemcolumnwidth`. Unfortunately I am a bit stuck here in this view at the moment. But maybe there you could come into play with a fresh new sight: Which kind of comment would have helped you here? I am glad to insert one, but right at the moment I am a bit lost what to write for clarification. 😅

Actually, after replacing `-\separatorcolumnwidth-\separatorcolumnwidth-\separatorcolumnwidth` by `3\separatorcolumnwidth` and such, I think the intent is much clearer now and it doesn't need a further comment.

I have approved the PR as-is now; feel free to merge it!  Many thanks for working on this!

LLdaniel replied:
Alright, then let us leave it as it is now. Thats also fine for me. 
Thank you for your code review and feedback! 👍

LLdaniel commented:
> LGTM, @LLdaniel ! After you checked that, there's nothing else to do here I think, right?

> Thanks for working on this!

Yes, now I do not know of additional points we should watch out for. I think everything is done with this checked list 👍

> EDIT: oh, GitHub UI is signalling a trivial-to-resolve conflict in `moderncvbodyiv.sty.`

You are right that is now my own change, because if I remember correctly: This was the PR which reminded me of the bug with the wrong initialization of the `\listitemsymbolspace`. Therefore I corrected it for your new "contemporary" style when I added it. This conflict should now be resolved with my last commit [20cc370](https://github.com/moderncv/moderncv/pull/177/commits/20cc3703d953a317123ca12105cf03251b909c7e).
